src/backend/distributed/commands/function.c

```diff
@@ -157,3 +161,90 @@ EnsureSequentialModeForFunctionDDL(void)
 						 "use only one connection for all future commands")));
 	SetLocalMultiShardModifyModeToSequential();
 }
+
+
+/*
+ * ShouldPropagateAlterFunction returns, based on the address of a function, if alter
+ * statements targeting the function should be propagated.
+ */
+static bool
+ShouldPropagateAlterFunction(const ObjectAddress *address)
+{
+	if (creating_extension)
+	{
+		/*
+		 * extensions should be created separately on the workers, functions cascading
+		 * from an extension should therefor not be propagated.
+		 */
+		return false;
+	}
+
+	if (!EnableDependencyCreation)
+	{
+		/*
+		 * we are configured to disable object propagation, should not propagate anything
+		 */
+		return false;
+	}
+
+	if (!IsObjectDistributed(address))
+	{
+		/* do not propagate alter function for non-distributed functions */
+		return false;
+	}
+
+	return true;
+}
+
+
+/*
+ * PlanAlterFunctionStmt is invoked for alter function statements with actions. Here we
+ * plan the jobs to be executed on the workers for functions that have been distributed in
+ * the cluster.
+ */
+List *
+PlanAlterFunctionStmt(AlterFunctionStmt *stmt, const char *queryString)
+{
+	const char *sql = NULL;
+	const ObjectAddress *address = NULL;
+	List *commands = NIL;
+
+	address = GetObjectAddressFromParseTree((Node *) stmt, false);
+	if (!ShouldPropagateAlterFunction(address))
+	{
+		return NIL;
+	}
+
+	EnsureCoordinator();
+
+	/* reconstruct alter statement in a portable fashion */
+	QualifyTreeNode((Node *) stmt);
+	sql = DeparseTreeNode((Node *) stmt);
+
+	/*
+	 * all functions that are distributed will need their alter statements propagated
+	 * regardless if in a transaction or not. If we would not propagate the alter
+	 * statement the functions would be different on worker and coordinator.
+	 */
+	EnsureSequentialModeForFunctionDDL();
+
+	commands = list_make3(DISABLE_DDL_PROPAGATION,
+						  (void *) sql,
+						  ENABLE_DDL_PROPAGATION);
+
+	return NodeDDLTaskList(ALL_WORKERS, commands);
+}
+
+
+const ObjectAddress *
+AlterFunctionStmtObjectAddress(AlterFunctionStmt *stmt, bool missing_ok)
```

Comment:
Needs a function comment. We do not expect readers to open this file with full knowledge of the dependency system.

Reply:
Will go through and comment all functions